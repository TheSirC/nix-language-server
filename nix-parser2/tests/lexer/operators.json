[
  {
    "kind": "Add",
    "span": {
      "start": 0,
      "end": 1
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 1,
      "end": 2
    }
  },
  {
    "kind": "Sub",
    "span": {
      "start": 2,
      "end": 3
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 3,
      "end": 4
    }
  },
  {
    "kind": "Mul",
    "span": {
      "start": 4,
      "end": 5
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 5,
      "end": 6
    }
  },
  {
    "kind": "Div",
    "span": {
      "start": 6,
      "end": 7
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 7,
      "end": 8
    }
  },
  {
    "kind": "IsEq",
    "span": {
      "start": 8,
      "end": 10
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 10,
      "end": 11
    }
  },
  {
    "kind": "NotEq",
    "span": {
      "start": 11,
      "end": 13
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 13,
      "end": 14
    }
  },
  {
    "kind": "LessThan",
    "span": {
      "start": 14,
      "end": 15
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 15,
      "end": 16
    }
  },
  {
    "kind": "LessThanEq",
    "span": {
      "start": 16,
      "end": 18
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 18,
      "end": 19
    }
  },
  {
    "kind": "GreaterThan",
    "span": {
      "start": 19,
      "end": 20
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 20,
      "end": 21
    }
  },
  {
    "kind": "GreaterThanEq",
    "span": {
      "start": 21,
      "end": 23
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 23,
      "end": 24
    }
  },
  {
    "kind": "LogicalAnd",
    "span": {
      "start": 24,
      "end": 26
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 26,
      "end": 27
    }
  },
  {
    "kind": "LogicalOr",
    "span": {
      "start": 27,
      "end": 29
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 29,
      "end": 30
    }
  },
  {
    "kind": "Concat",
    "span": {
      "start": 30,
      "end": 32
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 32,
      "end": 33
    }
  },
  {
    "kind": "Update",
    "span": {
      "start": 33,
      "end": 35
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 35,
      "end": 36
    }
  },
  {
    "kind": "Question",
    "span": {
      "start": 36,
      "end": 37
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 37,
      "end": 38
    }
  },
  {
    "kind": "Sub",
    "span": {
      "start": 38,
      "end": 40
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 40,
      "end": 41
    }
  },
  {
    "kind": "Not",
    "span": {
      "start": 41,
      "end": 42
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 42,
      "end": 43
    }
  }
]