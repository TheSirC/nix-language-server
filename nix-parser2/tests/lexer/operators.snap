Kind: Add, Span: [0, 1), Text: "+"
Kind: Whitespace, Span: [1, 2), Text: " "
Kind: Sub, Span: [2, 3), Text: "-"
Kind: Whitespace, Span: [3, 4), Text: " "
Kind: Mul, Span: [4, 5), Text: "*"
Kind: Whitespace, Span: [5, 6), Text: " "
Kind: Div, Span: [6, 7), Text: "/"
Kind: Whitespace, Span: [7, 8), Text: " "
Kind: IsEq, Span: [8, 10), Text: "=="
Kind: Whitespace, Span: [10, 11), Text: " "
Kind: NotEq, Span: [11, 13), Text: "!="
Kind: Whitespace, Span: [13, 14), Text: " "
Kind: LessThan, Span: [14, 15), Text: "<"
Kind: Whitespace, Span: [15, 16), Text: " "
Kind: LessThanEq, Span: [16, 18), Text: "<="
Kind: Whitespace, Span: [18, 19), Text: " "
Kind: GreaterThan, Span: [19, 20), Text: ">"
Kind: Whitespace, Span: [20, 21), Text: " "
Kind: GreaterThanEq, Span: [21, 23), Text: ">="
Kind: Whitespace, Span: [23, 24), Text: " "
Kind: LogicalAnd, Span: [24, 26), Text: "&&"
Kind: Whitespace, Span: [26, 27), Text: " "
Kind: LogicalOr, Span: [27, 29), Text: "||"
Kind: Whitespace, Span: [29, 30), Text: " "
Kind: Concat, Span: [30, 32), Text: "++"
Kind: Whitespace, Span: [32, 33), Text: " "
Kind: Update, Span: [33, 35), Text: "//"
Kind: Whitespace, Span: [35, 36), Text: " "
Kind: Question, Span: [36, 37), Text: "?"
Kind: Whitespace, Span: [37, 38), Text: " "
Kind: Imply, Span: [38, 40), Text: "->"
Kind: Whitespace, Span: [40, 41), Text: " "
Kind: Not, Span: [41, 42), Text: "!"
Kind: Whitespace, Span: [42, 43), Text: "\n"